core/src/main/codegen/templates/Parser.jj

```diff
@@ -2351,7 +2399,16 @@ SqlNode TableFunctionCall(SqlParserPos pos) :
             funcType = SqlFunctionCategory.USER_DEFINED_TABLE_SPECIFIC_FUNCTION;
         }
     ]
-    call = NamedRoutineCall(funcType, ExprContext.ACCEPT_CURSOR)
+    (
+        {
+            s = span();
+        }
+        // Currenly, only session window table function requires input to be set semantics
+        <SESSION>
+        call = SetSemanticsWindowTVF(new SqlIdentifier(Collections.singletonList("SESSION"), s.pos()), funcType)
+    |
```

Comment:
Hi, @julianhyde , do you think we should have this separate branch for session window parsing ?

Reply:
Hi @julianhyde I have same concern with @danny0405 , should we have separate branch for set semantic window? Now, only session windowing has set semantics.